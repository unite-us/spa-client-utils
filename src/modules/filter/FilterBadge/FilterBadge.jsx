import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Badge } from '@unite-us/ui';

const FilterBadge = ({ className, ...props }) => (
  <Badge
    className={classNames('filter-badge', className)}
    {...props}
  />
);

FilterBadge.propTypes = {
  /** Class name, in addition to `.filter-badge` */
  className: PropTypes.string,
  /** Badge background color, defaults to `white` */
  color: PropTypes.string,
  /** Badge shape, defaults to circle. */
  shape: PropTypes.string,
  /** Badge content, defaults to empty string */
  text: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

FilterBadge.defaultProps = {
  className: '',
  color: 'white',
  shape: 'circle',
  text: '',
};

export default FilterBadge;
