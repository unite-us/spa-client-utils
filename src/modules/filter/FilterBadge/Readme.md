```jsx
<FilterBadge text="A Filter Badge" />
<FilterBadge className="ml-one mr-one" text="short" />
<FilterBadge text="Looooooong Badge" />
```
