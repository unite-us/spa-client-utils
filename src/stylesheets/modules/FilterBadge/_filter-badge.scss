.filter-badge {
  border: solid 1px $display-color;
  border-radius: 100px;
  height: 25px;
}
