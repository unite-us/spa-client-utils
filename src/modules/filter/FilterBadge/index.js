import FilterBadge from './FilterBadge';

export default FilterBadge;
