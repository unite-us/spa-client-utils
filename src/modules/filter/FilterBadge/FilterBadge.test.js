import React from 'react';
import renderer from 'react-test-renderer';
import FilterBadge from './FilterBadge';

describe('FilterBadge', () => {
  it('renders a white badge with text', () => {
    expect(renderer.create(
      <FilterBadge text="Sheriff's Badge" />,
    ).toJSON()).toMatchSnapshot();
  });

  it('renders a blue (background) badge', () => {
    expect(renderer.create(
      <FilterBadge color="blue" text="Deputy's Badge" />,
    ).toJSON()).toMatchSnapshot();
  });

  it('returns null when no text passed', () => {
    expect(renderer.create(<FilterBadge />).toJSON()).toMatchSnapshot();
  });
});
