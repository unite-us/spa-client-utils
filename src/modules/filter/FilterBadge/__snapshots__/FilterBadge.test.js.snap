// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBadge renders a blue (background) badge 1`] = `
<span
  aria-label=""
  className="badge filter-badge"
  id=""
  style={
    Object {
      "backgroundColor": "blue",
      "color": "#4A657F",
    }
  }
>
  Deputy's Badge
</span>
`;

exports[`FilterBadge renders a white badge with text 1`] = `
<span
  aria-label=""
  className="badge filter-badge"
  id=""
  style={
    Object {
      "backgroundColor": "white",
      "color": "#4A657F",
    }
  }
>
  Sheriff's Badge
</span>
`;

exports[`FilterBadge returns null when no text passed 1`] = `null`;
